core/src/banking_stage.rs

```diff
@@ -1111,16 +1221,28 @@ impl BankingStage {
         transaction_status_sender: Option<TransactionStatusSender>,
         gossip_vote_sender: &ReplayVoteSender,
         qos_service: &QosService,
-    ) -> (usize, Vec<usize>) {
+    ) -> ProcessTransactionsSummary {
         let mut chunk_start = 0;
-        let mut unprocessed_txs = vec![];
-
+        let mut all_retryable_tx_indexes = vec![];
+        // All the transactions that attempted execution.
+        // These transactions can result in three cases after attempting execution:
+        // 1) Did not execute due to some fatal error like too old, or duplicate signature.
+        // 2) Were executed and committed, captured by `total_committed_transactions_count` below.
+        // 3) Were executed and failed commit, captured by `total_failed_commit_count` below.
+        let mut total_transactions_attempted_execution_count: usize = 0;
```

Comment:
Yeah, was trying to maintain parity with this original output `tx_len` here https://github.com/solana-labs/solana/blob/master/core/src/banking_stage.rs#L1339. I think it's safe to assume here that `tx_len` when used here: https://github.com/solana-labs/solana/blob/master/core/src/banking_stage.rs#L532, can just be replaced with `original_unprocess_indexes.len()`?

Reply:
Yeah, so I think `original_unprocess_indexes.len() - new_unprocessed_indexes.len()` would be become the number consumed (should be equal to what you had as `total_transactions_attempted_execution_count`). We agreed to use a separate field for the conditional here: https://github.com/solana-labs/solana/blob/master/core/src/banking_stage.rs#L532

> On the other hand, maybe it's safer to rely on the inner function to tell you how many transactions were executed in case something changes in the future (multiple transactions per packet for instance)?

Fair but this function already explicitly maps transactions to packets with the indexes vec anyways.

Reply:
> original_unprocess_indexes.len() - new_unprocessed_indexes.len() would be become the number consumed (should be equal to what you had as

Makes sense, switched here: https://github.com/solana-labs/solana/pull/22802/commits/ec71d532eacb274650aa1d3f0b602c446b7c5095

> Fair but this function already explicitly maps transactions to packets with the indexes vec anyways.

Good point

Last thing then. In my new new slot-based metrics I wanted to report the total number of transactions that attempted execution. For this case, you think it's worth reporting `total_transactions_attempted_execution_count` vs `original_unprocess_indexes.len()` to account for the case where at the end of the block some transactions weren't actually executed, they just get appended to the retryable set.

https://github.com/solana-labs/solana/blob/6d5bbca630bd59fb64f2bc446793c83482d8fba4/core/src/banking_stage.rs#L1140-L1152

These would be counted in ``original_unprocess_indexes.len()` but not in `total_transactions_attempted_execution_count`. This is actually one more good reason to switch to https://github.com/solana-labs/solana/pull/22802/commits/ec71d532eacb274650aa1d3f0b602c446b7c5095

Reply:
What insight are you trying to capture with "the total number of transactions that attempted execution"?

Reply:
was thinking the total number of txs entering the pipeline, but this is already captured by `committed` and `retried`, so we can remove this